src/main/java/org/opensearch/security/privileges/SystemIndexAccessEvaluator.java

```diff
@@ -302,6 +302,31 @@ private void evaluateSystemIndicesAccess(
                 }
         }
 
+        if (user.isPluginUser()) {
+            Set<String> matchingSystemIndices = SystemIndexRegistry.matchesPluginSystemIndexPattern(
+                user.getName().replace("plugin:", ""),
```

Comment:
what happens in-case of a preceding dot to the name of the system index but there is no dot in plugin name?

Will this always be a canonical class name of the plugin?

Reply:
Plugin user is only an in-memory construct which exists when a plugin executes code with its subject i.e. `pluginSubject.runAs(() -> { ...run actions here... });` (this is the replacement to `try (ThreadContext.StoredContext ctx = threadContext.stashContext()) { ... }` ).

By convention the username for a plugin user is `plugin:<canonicalClassName>`. i.e. for the security plugin it would be `plugin:org.opensearch.security.OpenSearchSecurityPlugin`. 

`SystemIndexRegistry.matchesPluginSystemIndexPattern` uses the canonical class name when determining the relationship between plugin -> system indices registered by the plugin using `SystemIndexRegistry.getSystemIndexDescriptors`